tests/client/core/api/test_api.js

```diff
@@ -7,18 +7,26 @@ import { ErrorHandler } from 'core/errorHandler';
 import { unexpectedSuccess } from 'tests/client/helpers';
 
 
+const defaultHeaders = {
+  'Content-Type': 'application/json',
+};
+
+function generateHeaders(headerData = defaultHeaders) {
+  return {
+    get: (name) => headerData[name],
```

Comment:
You could use a real `Response` object and return its `headers`. In fact we should probably use a real `Response` object in `createApiResponse` but one step at a time is probably fine.

Note: This won't identify the Firefox issue since we're running against release which is 51 but after the next update I think these tests will start to fail if we use a real `Response` (which is good).